src/Entity/Int32EntityId.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Wikibase\DataModel\Entity;
+
+use InvalidArgumentException;
+
+/**
+ * Interface for EntityIds that can be converted to a unique positive 32 bit integer in the range
+ * [1..2147483647], and back from the entity type and the number. Entity types that do not meet this
```

Comment:
If the range is [1..2147483647], the interface should be called Int31EntityId, right?
MySQL supports the full 32 bit unsigned integer range. If we want to support 32bit PHP, we'll need to limit ourselves to 31 bits, though. Postgres also doesn't seem to have an unsigned 4 byte int type, so 31 bit is safer.